python/tests/hypothesis/arcticdb/test_resample.py

```diff
@@ -0,0 +1,140 @@
+    columns = [hs_pd.column(name=f"col_{dtype}", dtype=dtype) for dtype in COLUMN_DTYPE]
+    result = draw(hs_pd.data_frames(columns, index=index))
+    result.sort_index(inplace=True)
+    return result
+
+@st.composite
+def origin(draw):
+    selected_origin = draw(st.sampled_from(["start", "end", "start_day", "end_day", "epoch", "timestamp"]))
+    # Hypothesis may generate dates for year > 2200 and some of the arithmetic operation will overflow.
+    if selected_origin == "timestamp":
+        min_date = MIN_DATE - np.timedelta64(365, 'D')
+        max_date = MAX_DATE + np.timedelta64(365, 'D')
+        return pd.Timestamp(draw(date(min_date=min_date, max_date=max_date)))
+    else:
+        return selected_origin
+
+def freq_fits_in_64_bits(count, unit):
+    """
+    This is used to check if a frequency is usable by Arctic. ArcticDB converts the frequency to signed 64 bit integer. 
+    """
+    billion = 1_000_000_000
+    mult = {'h': 3600 * billion, 'min': 60 * billion, 's': billion}
+    return (mult[unit] * count).bit_length() <= 63
+
+@st.composite
+def rule(draw):
+    count = draw(st.integers(min_value=1))
+    unit = draw(st.sampled_from(['min', 'h']))
+    result = f"{count}{unit}"
+    assume(freq_fits_in_64_bits(count=count, unit=unit))
+    return result
+
+@st.composite
+def offset(draw):
+    unit = draw(st.sampled_from(['s', 'min', 'h', None]))
+    if unit is None:
+        return None
+    count = draw(st.integers(min_value=1))
+    result = f"{count}{unit}"
+    assume(freq_fits_in_64_bits(count=count, unit=unit))
+    return result
+
+def drop_inf_and_nan(df):
+    return df[~df.isin([np.nan, np.inf, -np.inf]).any(axis=1)]
+
+def pd_resample_and_drop_empty_buckets(df, rule, offset, closed, label, origin, agg):
+    # Arctic drops empty buckets but Pandas does not. Add a column used only to track the number elements in the bucket.
+    # If its count is 0 the row is dropped.
+    count_col = '_count_col_'
+    count_df = pd.DataFrame({count_col: np.zeros(len(df.index), dtype="int")}, index=df.index)
+    to_resample = pd.concat([df, count_df], axis=1)
+    resampled = to_resample.resample(rule, closed=closed, label=label, offset=offset, origin=origin).agg(None, **agg, _count_col_=(count_col, "count"))
+    resampled = resampled[resampled[count_col] > 0]
+    resampled.drop(columns=[count_col], inplace=True)
+    return resampled
+
+def assert_resampled_dfs_approximate(left, right):
+    """
+    Checks if integer columns are exactly the same. For float columns checks if they are approximately the same.
+    We can't guarantee the same order of operations for the floats thus numerical errors might appear.
+    """
+    assert left.shape == right.shape
+    assert left.columns.equals(right.columns)
+    # To avoid checking the freq member of the index as arctic does not fill it in
+    assert list(left.index) == list(right.index)
```

Comment:
Might be more efficient to just check that the underlying ndarrays are the same?